src/lineargradientbrush.c

```diff
@@ -555,23 +555,30 @@ GdipCreateLineBrushFromRectI (GDIPCONST GpRect *rect, ARGB color1, ARGB color2,
 	if (!rect || !lineGradient)
 		return InvalidParameter;
 
+	if (mode < LinearGradientModeHorizontal || mode > LinearGradientModeBackwardDiagonal)
+		return OutOfMemory;
+
 	rectf.X = rect->X;
 	rectf.Y = rect->Y;
 	rectf.Width = rect->Width;
 	rectf.Height = rect->Height;
 
-	/* FIXME: Check whether angle has to be scalable or not in case of lineargradient mode. */
 	return GdipCreateLineBrushFromRectWithAngle (&rectf, color1, color2,
-		get_angle_from_linear_gradient_mode (mode), FALSE, wrapMode, lineGradient);
+		get_angle_from_linear_gradient_mode (mode), TRUE, wrapMode, lineGradient);
```

Comment:
The angle has to be scalable - the tests pass against GDI+ but fail with libgdiplus without this change